mail/api_test.py

```diff
@@ -0,0 +1,55 @@
+"""
+Test cases for email API
+"""
+from unittest.mock import patch
+from django.test import TestCase
+
+from micromasters import settings
+from mail.api import MailgunClient
+
+
+@patch('requests.post')  # pylint: disable=missing-docstring
+class MailAPITests(TestCase):
+    def test_from_address(self, mock_post):  # pylint: disable=no-self-use
+        """
+        Test that the 'from' address for our emails is set correctly
+        """
+        # NOTE: Using patch.multiple to override settings values because Django's
+        # override_settings decorator fails to work for mysterious reasons
+        with patch.multiple('mail.api', MAILGUN_FROM_EMAIL='mailgun_from_email@example.com'):
```

Comment:
Instead of using `patch.multiple` you should use [override_settings](https://docs.djangoproject.com/en/1.10/topics/testing/tools/#django.test.override_settings) here and below. You can use it as a decorator or as a context manager like here